File: 01-Languages/Dart/05-Control-Flow/Solutions/solution.dart
String fizzbuzz(int n) {
  return switch (n) {
    _ when n % 15 == 0 => 'FizzBuzz',
    _ when n % 3 == 0 => 'Fizz',
    _ when n % 5 == 0 => 'Buzz',
    _ => n.toString(),
  };
}

void main() {
  for (var i = 1; i <= 15; i++) {
    print(fizzbuzz(i));
  }
}

File: 01-Languages/Ruby/05-Control-Flow/Solutions/solution.rb
# Solutions -- Control Flow

# Exercise 1
def fizzbuzz(n)
  case
  when (n % 15).zero? then "FizzBuzz"
  when (n % 3).zero?  then "Fizz"
  when (n % 5).zero?  then "Buzz"
  else n.to_s
  end
end
puts (1..20).map { |n| fizzbuzz(n) }.join(" ")

# Exercise 2
def leap_year?(year)
  return true if (year % 400).zero?
  return false if (year % 100).zero?
  (year % 4).zero?
end

[2000, 1900, 2024, 2023].each do |year|
  puts "#{year}: #{leap_year?(year)}"
end
